From 544a7bfd8ee27f6b912b168ad733ce02918d91a6 Mon Sep 17 00:00:00 2001
From: "Uwe L. Korn" <uwe.korn@quantco.com>
Date: Wed, 18 Dec 2024 20:56:12 +0100
Subject: [PATCH] Disable mock_service introspection

---
 libsecret/meson.build | 24 +-----------------------
 1 file changed, 1 insertion(+), 23 deletions(-)

diff --git a/libsecret/meson.build b/libsecret/meson.build
index 85ad68f..e487462 100644
--- a/libsecret/meson.build
+++ b/libsecret/meson.build
@@ -195,28 +195,6 @@ mock_service_lib = library('mock-service',
   include_directories: config_h_dir,
 )
 
-if get_option('introspection')
-  mock_service_gir = gnome.generate_gir(mock_service_lib,
-    sources: files('mock-service.c', 'mock-service.h'),
-    namespace: 'MockService',
-    nsversion: '0',
-    export_packages: 'mock-service-0',
-    includes: [ 'GObject-2.0', 'Gio-2.0' ],
-    header: 'libsecret/mock-service.h',
-  )
-
-  if get_option('vapi')
-    mock_service_vapi = gnome.generate_vapi('mock-service-0',
-      sources: mock_service_gir[0],
-      packages: [
-        'gobject-2.0',
-        'gio-2.0',
-        libsecret_vapi,
-      ],
-    )
-  endif
-endif
-
 # C tests
 test_names = [
   'test-attributes',
@@ -253,7 +231,7 @@ foreach _test : test_names
 endforeach
 
 # Tests with introspection
-if get_option('introspection')
+if get_option('introspection') and false
   # env to be used in tests that use the typelib,
   # to make sure they find the one for MockService
   test_typelib_env = environment()
