From 16094ca91d673da1727775e2ae92d940d76371d0 Mon Sep 17 00:00:00 2001
From: Nehal J Wani <nehaljw.kkd1@gmail.com>
Date: Sun, 13 Feb 2022 16:22:14 -0500
Subject: [PATCH 3/3] Add patch for making libtool honor -Wl,--as-needed

---
 build/ltmain.sh | 27 +++++++++++++++++++++++++--
 1 file changed, 25 insertions(+), 2 deletions(-)

diff --git a/build/ltmain.sh b/build/ltmain.sh
index 7f3523d..f130c39 100644
--- a/build/ltmain.sh
+++ b/build/ltmain.sh
@@ -7220,6 +7220,11 @@ func_mode_link ()
 	arg=$func_stripname_result
 	;;
 
+      -Wl,--as-needed)
+	deplibs="$deplibs $arg"
+	continue
+	;;
+
       -Wl,*)
 	func_stripname '-Wl,' '' "$arg"
 	args=$func_stripname_result
@@ -7526,6 +7531,7 @@ func_mode_link ()
 
     case $linkmode in
     lib)
+	as_needed_flag=
 	passes="conv dlpreopen link"
 	for file in $dlfiles $dlprefiles; do
 	  case $file in
@@ -7537,6 +7543,7 @@ func_mode_link ()
 	done
 	;;
     prog)
+	as_needed_flag=
 	compile_deplibs=
 	finalize_deplibs=
 	alldeplibs=false
@@ -7606,6 +7613,15 @@ func_mode_link ()
 	lib=
 	found=false
 	case $deplib in
+	-Wl,--as-needed)
+	  if test prog,link = "$linkmode,$pass" ||
+	     test lib,link = "$linkmode,$pass"; then
+	    as_needed_flag="$deplib "
+	  else
+	    deplibs="$deplib $deplibs"
+	  fi
+	  continue
+	  ;;
 	-mt|-mthreads|-kthread|-Kthread|-pthread|-pthreads|--thread-safe \
         |-threads|-fopenmp|-openmp|-mp|-xopenmp|-omp|-qsmp=*)
 	  if test prog,link = "$linkmode,$pass"; then
@@ -10023,6 +10039,13 @@ EOF
 	  test "X$libobjs" = "X " && libobjs=
 	fi
 
+	# A bit hacky. I had wanted to add \$as_needed_flag to archive_cmds instead, but that
+	# comes from libtool.m4 which is part of the project being built. This should put it
+	# in the right place though.
+	if test lib,link = "$linkmode,$pass" && test -n "$as_needed_flag"; then
+	  libobjs=$as_needed_flag$libobjs
+	fi
+
 	save_ifs=$IFS; IFS='~'
 	for cmd in $cmds; do
 	  IFS=$sp$nl
@@ -10255,8 +10278,8 @@ EOF
       compile_deplibs=$new_libs
 
 
-      func_append compile_command " $compile_deplibs"
-      func_append finalize_command " $finalize_deplibs"
+      func_append compile_command " $as_needed_flag $compile_deplibs"
+      func_append finalize_command " $as_needed_flag $finalize_deplibs"
 
       if test -n "$rpath$xrpath"; then
 	# If the user specified any rpath flags, then add them.
-- 
2.35.0

